### PR TITLE
add callback url to JSON submission

### PR DESCRIPTION
**Pull request by asmega**
- this URL is used by submitter to fetch JSON representation of the form
- not sure about casing. snake/camel?
- i still need to investigate the structure of the JSON and add another PR if it needs changing

### CODE REVIEW

File: lib/controller/page/type/page.summary/page.summary.controller.js [hunk 1]
```diff
@@ -12,7 +12,11 @@ const CONSTANTS = require('../../../../constants/constants')
 const {
   SERVICE_OUTPUT_EMAIL,
   SERVICE_OUTPUT_JSON_ENDPOINT,
-  SERVICE_OUTPUT_JSON_KEY
+  SERVICE_OUTPUT_JSON_KEY,
+  SERVICE_SLUG,
+  PLATFORM_ENV,
+  DEPLOYMENT_ENV,
+  PORT
```
Comment (asmega) [inline lib/controller/page/type/page.summary/page.summary.controller.js]: is this the correct PORT to use? or hardcode to 3000?

File: lib/controller/page/type/page.summary/page.summary.controller.js [hunk 2]
```diff
@@ -160,15 +164,18 @@ SummaryController.attachEmailSubmissions = async (submissionId, userData, submis
...
   }
 
+  const callbackUrl = `http://${SERVICE_SLUG}.formbuilder-services-${PLATFORM_ENV}-${DEPLOYMENT_ENV}:${PORT}/api/submitter/json/default/${userData.getUserId()}.json`
+
   const submission = {
     type: 'json',
     url: SERVICE_OUTPUT_JSON_ENDPOINT,
-    encryption_key: SERVICE_OUTPUT_JSON_KEY
+    encryption_key: SERVICE_OUTPUT_JSON_KEY,
+    callback_url: callbackUrl
```
Comment (asmega) [inline lib/controller/page/type/page.summary/page.summary.controller.js]: should these be snake or camel cased?

Comment (asmega) [inline lib/controller/page/type/page.summary/page.summary.controller.js]: is `callback_url` a good name for this?

File: lib/controller/page/type/page.summary/page.summary.controller.js (hunk 1 above)
Comment (solidgoldpig) [reply lib/controller/page/type/page.summary/page.summary.controller.js]: `PORT` is the correct port to use, yes

File: lib/controller/page/type/page.summary/page.summary.controller.js (hunk 2 above)
Comment (solidgoldpig) [reply lib/controller/page/type/page.summary/page.summary.controller.js]: camelCased for the js, snake_cased for the ruby-backed endpoint

Comment (solidgoldpig) [reply lib/controller/page/type/page.summary/page.summary.controller.js]: `callbackUrl` implies the javascript will run it after it's finished thee request.

I'd prefer `data_url` or even just `data`

File: lib/controller/page/type/page.summary/page.summary.controller.js [hunk 3]
```diff
@@ -160,15 +164,18 @@ SummaryController.attachEmailSubmissions = async (submissionId, userData, submis
   }
 }
 
-SummaryController.attachJsonSubmission = async (submissions) => {
+SummaryController.attachJsonSubmission = async (submissions, userData) => {
   if (SERVICE_OUTPUT_JSON_ENDPOINT === undefined) {
     return
   }
 
+  const callbackUrl = `http://${SERVICE_SLUG}.formbuilder-services-${PLATFORM_ENV}-${DEPLOYMENT_ENV}:${PORT}/api/submitter/json/default/${userData.getUserId()}.json`
```
Comment (solidgoldpig) [inline lib/controller/page/type/page.summary/page.summary.controller.js]: Wouldn't this be better to pass the `http://${SERVICE_SLUG}.formbuilder-services-${PLATFORM_ENV}-${DEPLOYMENT_ENV}` part in as an env var?

File: lib/controller/page/type/page.summary/page.summary.controller.js (hunk 2 above)
Comment (asmega) [reply lib/controller/page/type/page.summary/page.summary.controller.js]: i've renamed this to `data_url`

File: lib/controller/page/type/page.summary/page.summary.controller.js (hunk 3 above)
Comment (asmega) [reply lib/controller/page/type/page.summary/page.summary.controller.js]: I've now extracted this to constants

**Pull request merged**